pkg/account/oauth-proxy.go

```diff
@@ -0,0 +1,46 @@
+package account
+
+import (
+	"fmt"
+
+	"k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	"github.com/jaegertracing/jaeger-operator/pkg/apis/io/v1alpha1"
+)
+
+// OAuthProxy returns a service account representing a client in the context of the OAuth Proxy
+func OAuthProxy(jaeger *v1alpha1.Jaeger) *v1.ServiceAccount {
+	trueVar := true
+	return &v1.ServiceAccount{
+		TypeMeta: metav1.TypeMeta{
+			APIVersion: "v1",
+			Kind:       "ServiceAccount",
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      OAuthProxyAccountNameFor(jaeger),
+			Namespace: jaeger.Namespace,
+			Annotations: map[string]string{
+				"serviceaccounts.openshift.io/oauth-redirectreference.primary": getOAuthRedirectReference(jaeger),
```

Comment:
I started with that, but it felt weird to have `if openshift` all over the code base. 

The idea behind the current approach is that we'd be deciding about the OAuth Proxy based on the `enabled` flag, and let the controller deal with setting the appropriate flag value based on the platform. The `normalize` operation will set it to false if not on OpenShift, and default to true if on OpenShift (leaving it as is if explicitly set).

Reply:
Just thinking about future platforms - might be better to define the structure now to make it easier for other platforms to be added - as otherwise it could be quite messy to try to decouple the binary decisions between k8s and openshift at a later date.